ssl/t1_lib.c

```diff
@@ -513,8 +514,12 @@ static int tls1_check_ec_key(SSL *s,
     }
     if (!curve_id)
         return 1;
+
+    if (!s->server && !check_own_groups)
```

Comment:
Is this the right condition?  If this is an escape clause for servers that don't check own groups, then it should be:
```
if (s->server && !check_own_groups) {
}
```
Otherwise it looks like an escape clause for clients...

Reply:
It is an escape clause for clients. Clients never check the server's groups. If they don't check their own either then we finish early. The server side is handled in the change to the for loop below.

Actually we never expect a client to call this function with `check_own_groups` set to 0, but I put this in for completeness (in case we ever do in the future).